packages/arb-validator-core/message/message.go

```diff
@@ -71,7 +77,86 @@ func intValueToAddress(val value.IntValue) common.Address {
 	return address
 }
 
-func UnmarshalUnsent(typecode MessageType, messageVal value.Value, chain common.Address) (UnsentMessage, error) {
+func addressesToValue(addresses []common.Address) value.TupleValue {
+	var ret []value.Value
+	for _, address := range addresses {
+		ret = append(ret, addressToIntValue(address))
+	}
+	return ListToStackValue(ret)
+}
+
+func valueToIntValuess(val value.Value) ([]value.IntValue, error) {
+	rawVals, err := StackValueToList(val)
+	if err != nil {
+		return nil, err
+	}
+	ints := make([]value.IntValue, 0, len(rawVals))
+	for _, rawVal := range rawVals {
+		rawValInt, ok := rawVal.(value.IntValue)
+		if !ok {
+			return nil, errors.New("value inside stack must be an int")
+		}
+		ints = append(ints, rawValInt)
+	}
+	return ints, nil
+}
+
+func valueToAddresses(val value.Value) ([]common.Address, error) {
+	ints, err := valueToIntValuess(val)
+	if err != nil {
+		return nil, err
+	}
+	addresses := make([]common.Address, 0, len(ints))
+	for _, intVal := range ints {
+		addresses = append(addresses, intValueToAddress(intVal))
+	}
+	return addresses, nil
+}
+
+func valueToInts(val value.Value) ([]*big.Int, error) {
+	ints, err := valueToIntValuess(val)
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]*big.Int, 0, len(ints))
+	for _, intVal := range ints {
+		ret = append(ret, intVal.BigInt())
+	}
+	return ret, nil
+}
+
+func valueToUInt32s(val value.Value) ([]uint32, error) {
+	ints, err := valueToIntValuess(val)
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]uint32, 0, len(ints))
+	for _, intVal := range ints {
+		ret = append(ret, uint32(intVal.BigInt().Uint64()))
+	}
+	return ret, nil
+}
+
+func intsToValue(ints []*big.Int) value.TupleValue {
+	var ret []value.Value
+	for _, val := range ints {
+		ret = append(ret, value.NewIntValue(new(big.Int).Set((val))))
+	}
+	return ListToStackValue(ret)
+}
+
+func uint32sToValue(ints []uint32) value.TupleValue {
+	var ret []value.Value
+	for _, val := range ints {
+		ret = append(ret, value.NewIntValue(new(big.Int).SetUint64((uint64(val)))))
+	}
+	return ListToStackValue(ret)
+}
+
+// UnmarshalExecuted converts the given Arbitrum message value which is the
+// encoding of one of our value types for a VM, back into the original message
+// type
+func UnmarshalExecuted(typecode MessageType, messageVal value.Value, chain common.Address) (ExecutionMessage, error) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [line is 118 characters, out of limit 80](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)